editor/editor_inspector.cpp

```diff
@@ -3211,14 +3211,6 @@ void EditorInspector::_edit_request_change(Object *p_object, const String &p_pro
 }
 
 void EditorInspector::_edit_set(const String &p_name, const Variant &p_value, bool p_refresh_all, const String &p_changed_field) {
-	if (autoclear && editor_property_map.has(p_name)) {
```

Comment:
I think this check should be kept at your added code.

Before, this function won't touch check state if not `autoclear`. After, it always updates the check state if undoable, and never updates when not-undoable.

It's true that currently in the editor, a `EditorInspector` is either set to not use undo, or set to not use autoclear, so it does not seem to cause any trouble. But that fact is outside the scope of this class.

Reply:
Great, my guess was right, and `UndoRedo::MERGE_ALL` would solve the problem.